Libplanet.Store/Trie/Nodes/FullNode.cs

```diff
@@ -34,48 +34,21 @@ public FullNode SetChild(int index, INode childNode)
             return new FullNode(Children.SetItem(index, childNode));
         }
 
-        bool IEquatable<FullNode>.Equals(FullNode? other)
+        /// <inheritdoc cref="IEquatable{T}.Equals"/>
+        public bool Equals(FullNode? other)
         {
-            if (other is null)
-            {
-                return false;
-            }
-
             if (ReferenceEquals(this, other))
             {
-                return false;
+                return true;
             }
 
-            for (var i = 0; i < ChildrenCount; ++i)
-            {
-                if (other.Children[i] is null ^ Children[i] is null)
-                {
-                    return false;
-                }
-
-                if (other.Children[i] is null && Children[i] is null)
-                {
-                    return true;
-                }
-
-                if (other.Children[i]?.GetType() != Children[i]?.GetType())
-                {
-                    return false;
-                }
-
-                if (other.Children[i]?.GetHashCode() != Children[i]?.GetHashCode())
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return other is { } node &&
+                Children.Where((n, i) => n is { })
+                    .SequenceEqual(node.Children.Where((n, i) => n is { }));
```

Comment:
It can return `true` even if they are different.

```csharp
using System;
using System.Linq;
					
public class Program
{
	public static void Main()
	{
		var arrA = new int?[3] { null, 1, null };
		var arrB = new int?[3] { 1, null, null };
		Console.WriteLine(arrA.Where((n, i) => n is { }).SequenceEqual(arrB.Where((n, i) => n is { }))); // True
	}
}

```

![image](https://github.com/planetarium/libplanet/assets/26626194/4e4b4f62-0105-4b2c-8293-ab785976a21c)

Reply:
My bad, I meant to compare *with* index. I'll fix it. 😱